tachyon/zk/plonk/vanishing/vanishing_utils.h

```diff
@@ -153,6 +157,65 @@ ExtendedPoly ExtendedToCoeff(const ExtendedEvals& evals,
   return poly;
 }
 
+template <typename Domain, typename Poly,
+          typename Evals = typename Domain::Evals, typename F>
+Evals CoeffToExtendedPart(const Domain* domain,
+                          const BlindedPolynomial<Poly>& poly, const F& zeta,
+                          const F& extended_omega_factor) {
+  Poly cloned = poly.poly();
+  Domain::DistributePowers(cloned, zeta * extended_omega_factor);
+  return domain->FFT(cloned);
+}
+
+template <typename Domain, typename Poly,
+          typename Evals = typename Domain::Evals, typename F>
+Evals CoeffToExtendedPart(const Domain* domain, const Poly& poly, const F& zeta,
+                          const F& extended_omega_factor) {
+  Poly cloned = poly;
+  Domain::DistributePowers(cloned, zeta * extended_omega_factor);
+  return domain->FFT(cloned);
+}
+
+template <typename Domain, typename Poly,
+          typename Evals = typename Domain::Evals, typename F>
+std::vector<Evals> CoeffsToExtendedPart(const Domain* domain,
```

Comment:
```suggestion
template <typename Domain, typename Poly, typename F>
std::vector<Domain::Evals> CoeffsToExtendedPart(const Domain* domain,
```

Reply:
This is a meaningful comment, but currently, we are not strictly limiting the relationship between template arguments. We will have to discuss this part again later.

Reply:
https://github.com/kroma-network/tachyon/issues/210